internal/xds/translator/listener.go

```diff
@@ -254,3 +257,65 @@ func buildXdsDownstreamTLSSecret(listenerName string,
 		},
 	}, nil
 }
+
+func buildXdsUDPListener(clusterName string, udpListener *ir.UDPListener) (*listener.Listener, error) {
+	if udpListener == nil {
+		return nil, errors.New("udp listener is nil")
+	}
+
+	statPrefix := "udp"
```

Comment:
From the [UDP Stats docs](https://www.envoyproxy.io/docs/envoy/latest/configuration/listeners/udp_filters/udp_proxy#statistics):
> The UDP proxy filter emits both its own downstream statistics as well as many of the [cluster upstream statistics](https://www.envoyproxy.io/docs/envoy/latest/configuration/upstream/cluster_manager/cluster_stats#config-cluster-manager-cluster-stats) where applicable. The downstream statistics are rooted at udp.<stat_prefix>.

With that said, `statPrefix` will cause stats to be generated with a prefix of `udp.udp`. Maybe use `service` as the value? Thoughts @arkodg @lizan?

Reply:
good point @danehans , I think we followed a similar pattern for other protocols too, will raise a issue to track and fix for all